src/coordinatesystems.jl

```diff
@@ -17,44 +19,97 @@ function Polar(r, θ)
+
 Base.show(io::IO, x::Polar) = print(io, "Polar(r=$(x.r), θ=$(x.θ) rad)")
-Base.isapprox(p1::Polar, p2::Polar; kwargs...) = isapprox(p1.r, p2.r; kwargs...) && isapprox(p1.θ, p2.θ; kwargs...)
+Base.isapprox(p1::PolarType, p2::PolarType; kwargs...) = isapprox(p1.r, p2.r; kwargs...) && isapprox(angle(p1), angle(p2); kwargs...)
+
+"""
+`Polard{T,A}(r::T, θ::A)` - 2D polar coordinates (using degrees)
+"""
+struct Polard{T,A} <: PolarType
+    r::T
+    θ::A
+
+    Polard{T, A}(r, θ) where {T, A} = new(r, θ)
+end
+
+function Polard(r, θ)
+    r2, θ2 = promote(r, θ)
+
+    return Polard{typeof(r2), typeof(θ2)}(r2, θ2)
+end
+
+# get angle in radians
+angle(x::Polard) = deg2rad(x.θ)
+
+Base.show(io::IO, x::Polard) = print(io, "Polard(r=$(x.r), θ=$(x.θ)°)")
+
+@inline Base.convert(::Type{Polar}, p::PolarType) = Polar(p.r, angle(p))
+@inline Base.convert(::Type{Polard}, p::Polar) = Polard(p.r, rad2deg(angle(p)))
+
 
 "`PolarFromCartesian()` - transformation from `AbstractVector` of length 2 to `Polar` type"
-struct PolarFromCartesian <: Transformation; end
-"`CartesianFromPolar()` - transformation from `Polar` type to `SVector{2}` type"
+struct PolarFromCartesian{PT<:PolarType} <: Transformation; end
+PolarFromCartesian() = PolarFromCartesian{Polar}() # default is Polar
+"`PolardFromCartesian()` - transformation from `AbstractVector` of length 2 to `Polar` type"
+const PolardFromCartesian = PolarFromCartesian{Polard}
+
+"`CartesianFromPolar()` - transformation from `Polar` or `Polard` type to `SVector{2}` type"
 struct CartesianFromPolar <: Transformation; end
 
-Base.show(io::IO, trans::PolarFromCartesian) = print(io, "PolarFromCartesian()")
+Base.show(io::IO, trans::PolarFromCartesian{P}) where {P} = print(io, "PolarFromCartesian{$P}()")
 Base.show(io::IO, trans::CartesianFromPolar) = print(io, "CartesianFromPolar()")
 
-function (::PolarFromCartesian)(x::AbstractVector)
+function (::PolarFromCartesian{Polar})(x::AbstractVector)
     length(x) == 2 || error("Polar transform takes a 2D coordinate")
 
     Polar(hypot(x[1], x[2]), atan(x[2], x[1]))
 end
 
-function transform_deriv(::PolarFromCartesian, x::AbstractVector)
+function (::PolarFromCartesian{Polard})(x::AbstractVector)
+    length(x) == 2 || error("Polard transform takes a 2D coordinate")
+
+    Polard(hypot(x[1], x[2]), rad2deg(atan(x[2], x[1])))
+end
+
+function transform_deriv(::PolarFromCartesian{Polar}, x::AbstractVector{T}) where {T}
     length(x) == 2 || error("Polar transform takes a 2D coordinate")
 
     r = hypot(x[1], x[2])
     f = x[2] / x[1]
-    c = one(eltype(x))/(x[1]*(one(eltype(x)) + f*f))
+    c = one(T) / (x[1] * (one(T) + f*f))
+    @SMatrix [ x[1]/r    x[2]/r ;
+              -f*c       c      ]
+end
+function transform_deriv(::PolarFromCartesian{Polard}, x::AbstractVector{T}) where {T}
+    length(x) == 2 || error("Polard transform takes a 2D coordinate")
+
+    r = hypot(x[1], x[2])
+    f = x[2] / x[1]
+    c = rad2deg(one(T)) / (x[1] * (one(T) + f*f))
```

Comment:
If the only difference is in the `rad2deg` you could factor that into `_angle_scaling(Polard)` vs `_angle_scaling(Polar)`